### PR TITLE
feat: gossip client with failover

### PR DESCRIPTION
**Pull request by alanshaw**
This PR adds `http-failover` and `http-failover-grace` args to the gossip `client` command to show how the gossip client can be used in conjunction with the drand http client with a failover to HTTP for if/when gossip randomness messages are delayed.

This is an update of https://github.com/drand/drand/pull/374

### CODE REVIEW

**Comment by willscott** [comment]
when can we subsume to a single client? 😁

File: cmd/relay-gossip/main.go [hunk 1]
```diff
@@ -144,23 +157,40 @@ var clientCmd = &cli.Command{
 			return xerrors.Errorf("constructing host: %w", err)
 		}
 
-		c, err := client.NewWithPubsub(ps, nil, nil)
+		chainHash, err := hex.DecodeString(cctx.String("chain-hash"))
```
Comment (willscott) [inline cmd/relay-gossip/main.go:160]: can this not be optional?

File: cmd/relay-gossip/main.go [hunk 2]
```diff
@@ -144,23 +157,40 @@ var clientCmd = &cli.Command{
...
+		httpFailover := cctx.StringSlice("http-failover")
+
+		var c client.Watcher
+		// if we have http failover endpoints then use the drand HTTP client with pubsub option
+		if len(httpFailover) > 0 {
+			grace := cctx.Duration("http-failover-grace")
+			if grace == 0 {
+				grace = time.Second * 5
+			}
+			c, err = client.New(
```
Comment (willscott) [inline cmd/relay-gossip/main.go:174]: can we make an `[]client.Option` with appropriate options for the two cases, and only create the Client once?

File: cmd/relay-gossip/main.go (hunk 1 above)
Comment (alanshaw) [reply cmd/relay-gossip/main.go:160]: Soon! It's one of my tasks for this week.

File: cmd/relay-gossip/main.go (hunk 2 above)
Comment (alanshaw) [reply cmd/relay-gossip/main.go:174]: No, if you don't specify http failover URLs it doesn't use the drand client.

**Pull request merged**